### PR TITLE
feat(arweave): reintroduce arweave code

### PR DESCRIPTION
**Pull request by james-a-morris**
Recommit of https://github.com/across-protocol/relayer-v2/pull/1198 directly to master.

### CODE REVIEW

**Review by nicholaspai — APPROVED** [review]
Have you tested this locally?

**Comment by james-a-morris** [comment]
> Have you tested this locally?

Yep - here's a sample run just now: https://arweave.app/tx/Z001KxHMa4wPxBd7xgbIes4HhtPSYsBCW9jwJI9vonU

File: src/dataworker/DataworkerUtils.ts [hunk 1]
```diff
@@ -716,3 +717,39 @@ export function l2TokensToCountTowardsSpokePoolLeafExecutionCapital(
...
+      message: `Data already exists on Arweave with tag: ${tag}`,
+      hash: matchingTxns.map((txn) => txn.hash),
+    });
+  } else {
+    const hashTxn = await client.set(data, tag);
+    const [address, balance] = await Promise.all([client.getAddress(), client.getBalance()]);
+    logger.info({
+      at: "Dataworker#index",
+      message: `Persisted data to Arweave with transaction hash: ${hashTxn}`,
+      hash: hashTxn,
```
Comment (nicholaspai) [inline src/dataworker/DataworkerUtils.ts:750]: I've noticed this URL doesn't make the JSON very easy to dump, are there alternative block explorers?

Comment (nicholaspai) [reply src/dataworker/DataworkerUtils.ts:750]: This link seems useful: https://arweave.net/Z001KxHMa4wPxBd7xgbIes4HhtPSYsBCW9jwJI9vonU

grabbed from https://viewblock.io/arweave/tx/Z001KxHMa4wPxBd7xgbIes4HhtPSYsBCW9jwJI9vonU

Comment (james-a-morris) [reply src/dataworker/DataworkerUtils.ts:750]: That's fair. This is just a transaction hash - we can definitely connect it to a block explorer

Comment (james-a-morris) [reply src/dataworker/DataworkerUtils.ts:750]: https://github.com/across-protocol/relayer-v2/pull/1281/commits/2ec52702d69b0862ca956f0fbb64ba5818d60b1e

**Comment by james-a-morris** [comment]
@nicholaspai @pxrl added multicaller & persistence to a parallel call here https://github.com/across-protocol/relayer-v2/pull/1281/commits/074b31f0cead8b8b71ba68929e2220096f07c432

File: src/dataworker/DataworkerUtils.ts [hunk 2]
```diff
@@ -716,3 +717,45 @@ export function l2TokensToCountTowardsSpokePoolLeafExecutionCapital(
...
+    const [address, balance] = await Promise.all([client.getAddress(), client.getBalance()]);
+    const endTime = utils.getCurrentTime();
+    logger.info({
+      at: "Dataworker#index",
+      message: `Persisted data to Arweave with transaction hash: ${hashTxn}`,
+      hash: hashTxn,
+      explorerUrl: `https://arweave.net/${hashTxn}`,
+      address,
+      balance,
+      elapsedSecondsToComplete: endTime - startTime,
```
Comment (nicholaspai) [inline src/dataworker/DataworkerUtils.ts:758]: elapsed time should maybe not be posted to slack, maybe keep this and other more dev-info in a separate debug log

File: src/dataworker/index.ts [hunk 3]
```diff
@@ -144,7 +154,36 @@ export async function runDataworker(_logger: winston.Logger, baseSigner: Signer)
...
+            clients.arweaveClient,
+            bundleDataToPersist,
+            logger,
+            `bundles-${bundleDataToPersist.bundleBlockRanges}`
+          );
+        }
+      };
+
+      // We want to persist the bundle data to the DALayer *AND* execute the multiCall transaction queue
+      // in parallel. We want to have both of these operations complete, even if one of them fails.
```
Comment (nicholaspai) [inline src/dataworker/index.ts:172]: If one completes, we want both to complete. Especially if we rely on this bundle data in other parts of the stack

File: src/dataworker/DataworkerUtils.ts (hunk 2 above)
Comment (james-a-morris) [reply src/dataworker/DataworkerUtils.ts]: Resolved and done so in a debug log https://github.com/across-protocol/relayer-v2/pull/1281/commits/c1a537de83eecde39bd79a47fdb8e545c7534e05

File: src/dataworker/index.ts (hunk 3 above)
Comment (james-a-morris) [reply src/dataworker/index.ts:172]: Resolved offline - I'll immediately follow up to make this resistant to AR failures via a new PR

**Pull request merged**